### PR TITLE
Support NetworkManager

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/<ticket-number>) |

#### Main changes
- Add support for `NetworkManager` by using the `nmcli` python library.
- Add separate backends to handle connections using wpa supplicant or network manager. Wifi manager decides the backend to use based on the system running the app.
- Add new app for advanced wi-fi settings in the Wi-Fi page (`nm-connection-editor`).
- Remove `pt-os-web-portal-wpa-gui-vnc` systemd service to instead start the GUI apps using `pt-web-vnc` directly.
- A lot of file/route renaming because of now supporting 2 advanced wi-fi gui apps (e.g.: `getVncWpaGuiUrl` -> `getVncAdvancedWifiGuiUrl`)
- Add tests.

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: pt_os_web_portal/backend/helpers/vnc_advanced_wifi_gui.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    # Determine which advanced wifi connection app to run
+    cmd = "systemctl is-active dhcpcd"
+    if run_command(cmd, check=False, timeout=5).strip() == "active":
+        app = "wpa_gui"
+    else:
+        app = "nm-connection-editor"
+
+    # Start pt-web-vnc
+    logging.info(f"Starting advanced wifi GUI with '{app}'")
+    start(
```
Comment (angusjfw) [inline pt_os_web_portal/backend/helpers/vnc_advanced_wifi_gui.py:37]: slight issue in that this will block, which may block all server requests (unless this uses a concurrent server of some sort eg wsgi)

**Review by angusjfw — COMMENTED** [review]
seems good but let's switch back to systemd for the advanced wifi gui and then need to manually test both os
